iothub/device/src/Pipeline/MqttTransportHandler.cs

```diff
@@ -1181,120 +1176,100 @@ private void HandleTwinResponse(MqttApplicationMessageReceivedEventArgs received
-                        Logging.Info(this, $"Received response to patch twin request with request id {receivedRequestId}.");
 
-                    IotHubClientErrorResponseMessage errorResponse = null;
+                    if (twinOperation.TwinPatchTask != null)
+                    {
+                        IotHubClientErrorResponseMessage error = status == 204
+                            ? null
+                            : ParseError(payloadBytes);
+
+                        // This received message is in response to an update reported properties request.
+                        var patchTwinResponse = new PatchTwinResponse
+                        {
+                            Status = status,
+                            Version = version,
+                            ErrorResponseMessage = error,
+                        };
 
-                    // This will only ever contain an error message which is encoded based on service contract (UTF-8).
-                    if (payloadBytes.Length > 0)
+                        twinOperation.TwinPatchTask.TrySetResult(patchTwinResponse);
+                    }
+                    else // should be a "get twin" operation
                     {
-                        string errorResponseString = Encoding.UTF8.GetString(payloadBytes);
-                        try
+                        var getTwinResponse = new GetTwinResponse
+                        {
+                            Status = status,
+                        };
+
+                        if (status != 200)
                         {
-                            errorResponse = DefaultPayloadConvention.Instance.GetObject<IotHubClientErrorResponseMessage>(errorResponseString);
+                            getTwinResponse.ErrorResponseMessage = ParseError(payloadBytes);
+                            twinOperation.TwinResponseTask.TrySetResult(getTwinResponse);
                         }
-                        catch (JsonException ex)
+                        else
                         {
-                            if (Logging.IsEnabled)
-                                Logging.Error(this, $"Failed to parse twin patch error response JSON. Message body: '{errorResponseString}'. Exception: {ex}. ");
-
-                            errorResponse = new IotHubClientErrorResponseMessage
+                            try
                             {
-                                Message = errorResponseString,
-                            };
+                                // Use the encoder that has been agreed to between the client and service to decode the byte[] response
+                                // The response is deserialized into an SDK-defined type based on service-defined NewtonSoft.Json-based json property name.
+                                // For this reason, we use Newtonsoft.Json serializer for this deserialization.
+                                TwinDocument clientTwinProperties = status == 200
```

Comment:
status is always 200 here

Reply:
Even if throttled?

Reply:
You have a check for status != 200 above and this is in the else block

Reply:
So all non 200 status code cases wouldn't reach here

Reply:
true that. will fix